src/test/groovy/hudson/plugins/gradle/injection/GradleSnippets.groovy

```diff
@@ -32,19 +32,19 @@ class GradleSnippets {
     static WorkflowJob pipelineJobWithCredentials(JenkinsRule j) {
         def pipelineJob = j.createProject(WorkflowJob)
 
-        pipelineJob.setDefinition(new CpsFlowDefinition("""
+        pipelineJob.setDefinition(new CpsFlowDefinition('''
     stage('Build') {
       node('foo') {
         withCredentials([string(credentialsId: 'my-creds', variable: 'PASSWORD')]) {
           if (isUnix()) {
-            sh "echo password=\$PASSWORD"
+            sh 'echo password=$PASSWORD'
```

Comment:
The original test was improperly interpolating the password in Groovy code on the controller, rather than using the environment variable, and so actually running the shell with e.g.

```sh
echo password=actual secret here
```

Pipeline prints a warning about this to the build log. Does not make much difference for purposes of the masking issue (since either way the shell step output contains the secret), but better to test the script that users are actually supposed to write.

Reply:
I did not see this warning in the test logs 🤔 but I understand the issue, quite sneaky!